src/ipz_parser.cpp

```diff
@@ -440,4 +444,142 @@ types::VPDMapVariant IpzVpdParser::parse()
     }
 }
 
+types::BinaryVector
+    IpzVpdParser::getKeywordValueFromRecord(const types::Record& i_record,
+                                            const types::Keyword& i_keyword,
+                                            auto i_recordDataOffset)
+{
+    // Go to the record's offset
+    auto l_iterator = m_vpdVector.cbegin();
+
+    std::ranges::advance(l_iterator, i_recordDataOffset, m_vpdVector.cend());
+
+    // skip large resource field.
+    std::ranges::advance(l_iterator, Length::LARGE_RESOURCE,
+                         m_vpdVector.cend());
+
+    // Get record length
+    auto l_recordLength = readUInt16LE(l_iterator);
+    // Get iterator to end of the record
+    auto l_recordEndItr = std::next(l_iterator, l_recordLength);
+
+    // Skip record length field
+    std::ranges::advance(l_iterator, Length::RECORD_LENGTH, m_vpdVector.cend());
+    while (l_iterator < l_recordEndItr)
+    {
+        const auto l_kwName =
+            std::string(l_iterator, std::next(l_iterator, Length::KW_NAME));
+
+        char l_kwNameStart = *l_iterator;
+
+        std::ranges::advance(l_iterator, Length::KW_NAME, m_vpdVector.cend());
+
+        auto l_kwdDataLength = 0;
+
+        if (constants::LAST_KW == l_kwName)
+        {
+            throw std::runtime_error("Given keyword not found.");
+        }
+
+        // Check if we process the right record
+        if (l_kwName == "RT")
+        {
+            l_kwdDataLength = *l_iterator;
+
+            // Get RT keyword's value which is the record name
+            std::ranges::advance(l_iterator, sizeof(types::KwSize),
+                                 m_vpdVector.cend());
+
+            if (i_record !=
+                std::string(l_iterator,
+                            std::next(l_iterator, Length::RECORD_NAME)))
+            {
+                throw std::runtime_error(
+                    "Given record is not present in the offset provided");
+            }
+
+            std::ranges::advance(l_iterator, l_kwdDataLength,
+                                 m_vpdVector.cend());
+            continue;
+        }
+
+        if (constants::POUND_KW == l_kwNameStart)
+        {
+            l_kwdDataLength = readUInt16LE(l_iterator);
+
+            // Jump past 2Byte keyword length for pound keyword
+            std::ranges::advance(l_iterator, sizeof(types::PoundKwSize),
+                                 m_vpdVector.cend());
+        }
+        else
+        {
+            l_kwdDataLength = *l_iterator;
+            std::ranges::advance(l_iterator, sizeof(types::KwSize),
+                                 m_vpdVector.cend());
+        }
+
+        if (l_kwName == i_keyword)
+        {
+            // Return keyword's value to the caller
+            const auto l_begin = l_iterator;
```

Comment:
nit: can avoid these kind of one time use local variable.

Reply:
this is not one time use variable. here am saving the starting point of the keyword's value in l_begin. so that in line 527, I can fetch the keyword value completely.

may be for clarity I'll change the variable name to l_kwValueStart.

Reply:
I meant, you can get rid of these kid of variables.

`const auto l_kwValueStart = l_iterator;
            std::ranges::advance(l_iterator, l_kwdDataLength,
                                 m_vpdVector.cend());
            return types::BinaryVector(l_kwValueStart, l_iterator);`
            
  can be written as 
  `return types::BinaryVector(l_iterator, std::ranges::next(l_iterator, l_kwdDataLength,
                                 m_vpdVector.cend()));`

Reply:
done